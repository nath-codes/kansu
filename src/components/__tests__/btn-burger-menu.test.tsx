import 'jest-styled-components'
import React from 'react'

import BtnBurgerMenu from '../btn-burger-menu'
import { renderWithTheme } from './helpers'

describe('BtnBurgerMenu', () => {
  it('renders correctly', () => {
    let tree = renderWithTheme(
      <BtnBurgerMenu isActive={false} handleClick={() => 'foo'} />
    ).toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderWithTheme(
      <BtnBurgerMenu isActive={true} handleClick={() => 'foo'} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
