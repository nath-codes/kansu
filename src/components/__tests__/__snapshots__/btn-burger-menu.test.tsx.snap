// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnBurgerMenu renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 2.25rem;
}

.c1 {
  background-color: #ffffff;
  height: 0.25rem;
  position: relative;
  width: 100%;
  -webkit-transition-duration: 0.075s;
  transition-duration: 0.075s;
  -webkit-transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
  transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
}

.is-active .c1 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-transition-delay: 0.12s;
  transition-delay: 0.12s;
  -webkit-transition-timing-function: cubic-bezier(0.215,0.61,0.355,1);
  transition-timing-function: cubic-bezier(0.215,0.61,0.355,1);
}

.c1:before {
  border-top: solid 0.25rem #ffffff;
  content: '';
  height: 0.25rem;
  left: 0;
  opacity: 1;
  position: absolute;
  right: 0;
  top: -0.625rem;
  -webkit-transition: top 0.075s 0.12s ease,opacity 0.075s ease;
  transition: top 0.075s 0.12s ease,opacity 0.075s ease;
}

.is-active .c1:before {
  opacity: 0;
  top: 0;
  -webkit-transition: top 0.075s ease,opacity 0.075s 0.12s ease;
  transition: top 0.075s ease,opacity 0.075s 0.12s ease;
}

.c1:after {
  bottom: -0.625rem;
  border-bottom: solid 0.25rem #ffffff;
  content: '';
  height: 0.25rem;
  left: 0;
  position: absolute;
  right: 0;
  opacity: 1;
  -webkit-transition: bottom 0.075s 0.12s ease, -webkit-transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
  -webkit-transition: bottom 0.075s 0.12s ease, transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
  transition: bottom 0.075s 0.12s ease, transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
}

.is-active .c1:after {
  bottom: 0;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: bottom 0.075s ease, -webkit-transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
  -webkit-transition: bottom 0.075s ease, transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
  transition: bottom 0.075s ease, transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
}

@media only screen and (min-width:768px) {
  .c0 {
    display: none;
  }
}

<button
  aria-label="Toggle Menu"
  className="c0"
  onClick={[Function]}
>
  <div
    className="c1"
  />
</button>
`;

exports[`BtnBurgerMenu renders correctly 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 2.25rem;
}

.c1 {
  background-color: #ffffff;
  height: 0.25rem;
  position: relative;
  width: 100%;
  -webkit-transition-duration: 0.075s;
  transition-duration: 0.075s;
  -webkit-transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
  transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
}

.is-active .c1 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-transition-delay: 0.12s;
  transition-delay: 0.12s;
  -webkit-transition-timing-function: cubic-bezier(0.215,0.61,0.355,1);
  transition-timing-function: cubic-bezier(0.215,0.61,0.355,1);
}

.c1:before {
  border-top: solid 0.25rem #ffffff;
  content: '';
  height: 0.25rem;
  left: 0;
  opacity: 1;
  position: absolute;
  right: 0;
  top: -0.625rem;
  -webkit-transition: top 0.075s 0.12s ease,opacity 0.075s ease;
  transition: top 0.075s 0.12s ease,opacity 0.075s ease;
}

.is-active .c1:before {
  opacity: 0;
  top: 0;
  -webkit-transition: top 0.075s ease,opacity 0.075s 0.12s ease;
  transition: top 0.075s ease,opacity 0.075s 0.12s ease;
}

.c1:after {
  bottom: -0.625rem;
  border-bottom: solid 0.25rem #ffffff;
  content: '';
  height: 0.25rem;
  left: 0;
  position: absolute;
  right: 0;
  opacity: 1;
  -webkit-transition: bottom 0.075s 0.12s ease, -webkit-transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
  -webkit-transition: bottom 0.075s 0.12s ease, transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
  transition: bottom 0.075s 0.12s ease, transform 0.075s cubic-bezier(0.55,0.055,0.675,0.19);
}

.is-active .c1:after {
  bottom: 0;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
  -webkit-transition: bottom 0.075s ease, -webkit-transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
  -webkit-transition: bottom 0.075s ease, transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
  transition: bottom 0.075s ease, transform 0.075s 0.12s cubic-bezier(0.215,0.61,0.355,1);
}

@media only screen and (min-width:768px) {
  .c0 {
    display: none;
  }
}

<button
  aria-label="Toggle Menu"
  className="is-active c0"
  onClick={[Function]}
>
  <div
    className="c1"
  />
</button>
`;
